gen.go

```diff
@@ -86,34 +86,80 @@ func New(injs ...func(cmd *kingpin.CmdClause)) *Generator {
 // Example:
 //
 // ```
-//  gen := gen.With("mycompany.com", "production", "eu1", "kubernetes", "thanos")
+//
+//	gen := gen.With("mycompany.com", "production", "eu1", "kubernetes", "thanos")
+//
 // ```
 // Giving the path `mycompany.com/production/eu1/kubernetes/thanos`.
 //
 // With return a Generator pointing at the specified path which can be specified even further:
 // Example:
 // ```
-//   gen := mimic.New()
-//   // gen/
-//   ...
-//   gen = gen.With('foo')
-//   // gen/foo
-//   ...
-//   {
-//     gen := gen.With('bar')
-//     // gen/foo/bar
-//   }
-//   // gen/foo
+//
+//	gen := mimic.New()
+//	// gen/
+//	...
+//	gen = gen.With('foo')
+//	// gen/foo
+//	...
+//	{
+//	  gen := gen.With('bar')
+//	  // gen/foo/bar
+//	}
+//	// gen/foo
+//
 // ```
 func (g *Generator) With(parts ...string) *Generator {
 	// TODO(bwplotka): Support "..", to get back?
 
 	return &Generator{
 		out: g.out,
 		FilePool: FilePool{
-			Logger: g.Logger,
-			path:   append(g.path, parts...),
-			m:      g.m,
+			Logger:           g.Logger,
+			path:             append(g.path, parts...),
+			m:                g.m,
+			topLevelComments: g.topLevelComments,
+		},
+	}
+}
+
+// WithTopLevelComment enables mimic to add any string as a header comment for a genenrated
+// file. Follows same usage semantics as With().
+//
+// Example:
+//
+//	gen := mimic.New()
+//	defer gen.Generate()
+//
+//	gen.With("config").WithTopLevelComment(mimic.GeneratedComment).Add(name+".yaml", encoding.GhodssYAML(config))
+//
+// Like With, you can also chain multiple WithTopLevelComment(), which will be added to the top of a file in that order.
+//
+// Example:
+//
+//	gen := mimic.New()
+//
+//	defer gen.Generate()
+//	gen = gen.WithTopLevelComment("Foo.")
+//
+//	gen.With("config").WithTopLevelComment("Bar.").Add(name+".yaml", encoding.GhodssYAML(config))
+//
+// This will result in a YAML file like,
+//
+//	# Foo.
+//	# Bar.
+//	config:
+//	- prometheus:
+//
+// NOTE: This option is not compatible with encoding.JSON and will be treated as a no-op, as it does not support comments.
+func (g *Generator) WithTopLevelComment(content ...string) *Generator {
```

Comment:
What about to be consistent with other APIs?

```suggestion
func (g *Generator) WithTopLevelComment(content string) *Generator {
```

Reply:
Yeah I was think of keeping this the same as `With`, but makes more sense